Add Ruby example project

aussiDavid opened the pull request:
I've added a Ruby example project with RSpec and Simplecov to generate lcov code coverage reports that are suitable for vscode-coverage-gutters.

ryanluker commented on example/ruby/README.md:
```diff
@@ -0,0 +1,32 @@
+require 'simplecov-lcov'
+
+SimpleCov::Formatter::LcovFormatter.config do |c|
+  c.report_with_single_file = true # vscode-coverage-gutters preffers a single file for lcov reporting
```
minor: sp* preffers -> prefers

ryanluker commented on example/ruby/README.md:
```diff
@@ -0,0 +1,32 @@
+
+## Usage
+
+That's it! I find this extention works well with [guard](https://github.com/guard/guard) and [guard-rspec](https://github.com/guard/guard-rspec) as it updates the gutters in real time when the test suite is executed.
```
minor: SP* extention -> extension

ryanluker reviewed (commented):
Looks great thanks for the contribution!

If you can commit the lcov file generated by the example project for the example code, I can make a new integration test that will use the new example to make sure we don't regress on ruby language projects 😁 .

https://github.com/ryanluker/vscode-coverage-gutters/blob/master/test/extension.test.ts#L161-L181

aussiDavid commented:
@ryanluker Thanks. I've added a `lcov.info` file and updated the `README.md` with additional information.

I tried to write the integration test myself but I could not get the test suite to run.

ryanluker commented:
@aussiDavid no problem, I checked out your branch and added an test just below the java one in `extension.test.ts` on line 182.
```typescript
    test("Run display coverage on ruby test file @integration", async () => {
        const decorationSpy = sinon.spy(Renderer.prototype, "setDecorationsForEditor");
        const extension = await vscode.extensions.getExtension("ryanluker.vscode-coverage-gutters");
        if (!extension) {
            throw new Error("Could not load extension");
        }

        const testCoverage = await vscode.workspace.findFiles("**/ruby/lib/app/math.rb", "**/node_modules/**");
        const testDocument = await vscode.workspace.openTextDocument(testCoverage[0]);
        await vscode.window.showTextDocument(testDocument);
        await vscode.commands.executeCommand("coverage-gutters.displayCoverage");

        await checkCoverage(() => {
            // Look for exact coverage on the ruby file
            const cachedLines: ICoverageLines = decorationSpy.getCall(0).args[1];
            expect(cachedLines.full).to.have.lengthOf(5);
            expect(cachedLines.none).to.have.lengthOf(0);
        });

        decorationSpy.restore();
    });
```

aussiDavid commented:
Out of curiosity, how do you run the tests? Is there something special that
needs to be installed/configured?

On Mon, 26 Jul 2021 at 14:22, Ryan Luker ***@***.***> wrote:

> @aussiDavid <https://github.com/aussiDavid> no problem, I checked out
> your branch and added an test just below the java one in extension.test.ts
> on line 182.
>
>     test("Run display coverage on ruby test file @integration", async () => {
>         const decorationSpy = sinon.spy(Renderer.prototype, "setDecorationsForEditor");
>         const extension = await vscode.extensions.getExtension("ryanluker.vscode-coverage-gutters");
>         if (!extension) {
>             throw new Error("Could not load extension");
>         }
>
>         const testCoverage = await vscode.workspace.findFiles("**/ruby/lib/app/math.rb", "**/node_modules/**");
>         const testDocument = await vscode.workspace.openTextDocument(testCoverage[0]);
>         await vscode.window.showTextDocument(testDocument);
>         await vscode.commands.executeCommand("coverage-gutters.displayCoverage");
>
>         await checkCoverage(() => {
>             // Look for exact coverage on the ruby file
>             const cachedLines: ICoverageLines = decorationSpy.getCall(0).args[1];
>             expect(cachedLines.full).to.have.lengthOf(5);
>             expect(cachedLines.none).to.have.lengthOf(0);
>         });
>
>         decorationSpy.restore();
>     });
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ryanluker/vscode-coverage-gutters/pull/335#issuecomment-886372628>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABMVYGGFTOYERE4NHHSZPRTTZTSXDANCNFSM5A5OV2DQ>
> .
>


-- 
David Milanese

ryanluker commented:
@aussiDavid there is a couple different ways to run the tests, the easiest is to use the headless mode in a command line (imo).
You could also use the built in tooling vscode gives extensions (and what we have defined in the config for debugging / running tests).

![Screenshot 2021-07-25 215925](https://user-images.githubusercontent.com/1335972/126935656-b62345a9-2b33-4009-b8db-6475cedef03d.png)
![Screenshot 2021-07-25 220114](https://user-images.githubusercontent.com/1335972/126935657-30477f3f-1ddb-4799-abf5-d553b69500cc.png)

ryanluker commented:
@aussiDavid Oh I forgot to mention but the github action workflow doesn't automatically run on your PRs until you have one successfully merged to master in this repository. In the meantime I just push a button so the tests can run on your code 😁.

aussiDavid commented:
Done. @ryanluker please run the test, I occasionally get a failed test when checking the contents of the code average preview file (index.html).

ryanluker commented:
@aussiDavid Looks great, thanks for adding a variety of test coverage into the lcov.info so that way the extension test can assert more types of coverage 👍🏻. I noticed with the first test I wrote I only could assert coverage or no coverage 🤔.
